src/reporting.nim

```diff
@@ -105,21 +105,24 @@ template doEmbeddedReport(): Box =
     pack[seq[Box]](@[])
 
 template doCustomReporting() =
-  for topic, spec in chalkConfig.reportSpecs:
-    if not spec.enabled: continue
+  for topic, spec in getChalkSubsections("custom_report"):
+    let enabledOpt = getOpt[bool](spec, "enabled")
+    if enabledOpt.isNone() or not enabledOpt.get(): continue
     var
-      sinkConfs = spec.sinkConfigs
+      sinkConfs = get[seq[string]](spec, "sink_configs")
 
     discard registerTopic(topic)
 
-    if getCommandName() notin spec.useWhen and "*" notin spec.useWhen:
+    let useWhen = get[seq[string]](spec, "use_when")
+    if getCommandName() notin useWhen and "*" notin useWhen:
       continue
     if topic == "audit" and not get[bool](chalkConfig, "publish_audit"):
       continue
     if len(sinkConfs) == 0 and topic notin ["audit", "chalk_usage_stats"]:
       warn("Report '" & topic & "' has no configured sinks.  Skipping.")
 
-    let templateToUse = chalkConfig.reportTemplates[spec.reportTemplate]
+    let reportTemplate = get[string](spec, "report_template")
+    let templateToUse = getObject(getChalkScope(), "report_template." & reportTemplate)
```

Comment:
nit

```suggestion
    let
      reportTemplate = get[string](spec, "report_template")
      templateToUse = getObject(getChalkScope(), "report_template." & reportTemplate)
```

chalk predominatly uses this pattern for defining multiple vars at once